src/distilabel/llm/litellm.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2023-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import TYPE_CHECKING, List, Optional, Union
+
+from pydantic import PrivateAttr
+
+from distilabel.llm.base import AsyncLLM
+
+if TYPE_CHECKING:
+    from distilabel.llm.typing import GenerateOutput
+    from distilabel.steps.task.typing import ChatType
+
+
+class LitellmLLM(AsyncLLM):
+    """LiteLLM LLM implementation running the async API client.
+
+    Attributes:
+        model: the model name to use for the LLM e.g. "gpt-3.5-turbo", "gpt-4", etc.
+        api_key: the API key to authenticate the requests to the OpenAI API.
+    """
+
+    model: str
+    litellm_logging: Optional[bool] = False
+    _aclient: Optional["callable"] = PrivateAttr(...)
+
+    def load(
+        self,
+    ) -> None:
+        """
+        Loads the `AsyncOpenAI` client to benefit from async requests.
+
+        Args:
+            model: the model name to use for the LLM e.g. "gpt-3.5-turbo", "gpt-4", etc.
```

Comment:
This docstring also needs an update

```suggestion
        Loads the `AsyncOpenAI` client to benefit from async requests.
```